### PR TITLE
Skip button update

### PR DESCRIPTION
**Pull request by Casulono**
# Description

The skip button is currently unintuitive as in reality, it doesn't skip levels that someone hasn't beaten yet and just returns to level select. This small PR changes it and the skip button allows skipping levels to locked ones. It will unlock them but won't mark them as complete, the miniature will still feature blank level when it is implemented for all of them.

## Related issue(s)

https://github.com/Saplings-Projects/Birthday-2024/issues/83

## List of changes

A more detailed list of the changes.
- removed the path that leads to the level select screen when skipping to the locked level
- now skip button will lead to the next level unless it's the last one
- skip button will unlock levels but won;t mark them as complete

### CODE REVIEW

**Comment by Mikotey** [comment]
My initial understanding was that the skip button was primarily for the sapling levels where none of them are locked. Skipping was mostly just to navigate to the next. Not so much to force progression

There were some additional changes to be made in order for campaign tracking to work as you intended. Previously, the campaign levels relied on the integer to track completion, so skipping did treat them as beaten and completed. Now the campaign levels use the same method as the sapling levels by checking the completion dictionary.
Since campaign levels don't have a level name, it couldn't rely on that as the key for tracking completion. I've changed the key to be the level setup resource path (was going to use resource name, but apparently that just defaults to empty string...). This key seems more reliable in the case that two levels have the same name

**Review by Mikotey — APPROVED** [review]
Apparently I can approve a PR after applying my own changes?

**Comment by Mikotey** [comment]
Nevermind. I guess not lol

File: Birthday-2024-Project/Resources/CampaignLevels/CampaignLevel1.tres
Comment (Casulono) [inline Birthday-2024-Project/Resources/CampaignLevels/CampaignLevel1.tres:4]: emotes as previews look very crusty, I'm for making PNG previews for campaign just like for Sapling Levels

**Comment by Casulono** [comment]
oh, and I also can't approve this PR

File: Birthday-2024-Project/Resources/CampaignLevels/CampaignLevel1.tres
Comment (Mikotey) [reply Birthday-2024-Project/Resources/CampaignLevels/CampaignLevel1.tres:4]: These are just placeholder until the proper level previews are added

**Comment by Mikotey** [comment]
@Samuel-Fox save us

**Review by Samuel-Fox — APPROVED** [review]

**Pull request merged**
